### PR TITLE
chore: bump to yarn 4

### PR DESCRIPTION
**Pull request by cpaulve-1A**
## Related issues

- :bug: Fixes #(issue)
- :rocket: Feature #(issue)

### CODE REVIEW

File: .yarnrc.yml [hunk 1]
```diff
@@ -1,10 +1,10 @@
+compressionLevel: mixed
+
+enableGlobalCache: false
```
Comment (kpanot) [inline .yarnrc.yml:3]: Why?
You not be only for the CI so be part of the `setup` workflow?

Comment (cpaulve-1A) [reply .yarnrc.yml:3]: Oups I didn't catch this one, it was done by the yarn set

Comment (cpaulve-1A) [reply .yarnrc.yml:3]: This was the previous configuration on yarn3 and changed to true in yarn4 - this part was done by their migration script.

File: tools/github-actions/audit/src/main.ts [hunk 2]
```diff
@@ -5,54 +5,148 @@ import {GitHubAdvisoryId, NPMAuditReportV1, Severity} from 'audit-types';
...
+ * Severities supported by yarn npm audit from the lowest to the highest criticality
+ */
+const severities: Severity[] = ['info', 'low', 'moderate', 'high', 'critical'];
+const colors = ['', 'green', 'yellow', 'orange', 'red'];
+
+/**
+ * Interface to describe Yarn 4+ npm audit response.
+ * It is not yet covered by audit-types
+ */
+interface Yarn4AuditResponse {
```
Comment (kpanot) [inline tools/github-actions/audit/src/main.ts:18]: Do you know it this types can be exposed by [@yarnpkg/types](https://www.npmjs.com/package/@yarnpkg/types) ?

Comment (cpaulve-1A) [reply tools/github-actions/audit/src/main.ts:18]: Nope, it seems more linked to methods and types around workspace management etc. there is nothing around audit on this one. I'll try to check in yarnpkg scope if there is anything exposed

Comment (cpaulve-1A) [reply tools/github-actions/audit/src/main.ts:18]: Nope I didn't find anything relevant

File: packages/@o3r/test-helpers/src/test-environments/create-test-environment-angular.ts [hunk 3]
```diff
@@ -140,7 +140,7 @@ export async function createTestEnvironmentAngular(inputOptions: Partial<CreateT
       const packageJson = JSON.parse(packageJsonString);
       writeFileSync(workspacePackageJsonPath, JSON.stringify({
         ...packageJson,
-        scripts: {...packageJson.scripts, build: getPackageManager() === 'npm' ? 'npm run build --workspaces' : 'yarn workspaces foreach run build'},
+        scripts: {...packageJson.scripts, build: getPackageManager() === 'npm' ? 'npm run build --workspaces' : 'yarn workspaces foreach -A run build'},
```
Comment (kpanot) [inline packages/@o3r/test-helpers/src/test-environments/create-test-environment-angular.ts:143]: the `-A` is also working with yarn3 + workspaces plugin?

Comment (cpaulve-1A) [reply packages/@o3r/test-helpers/src/test-environments/create-test-environment-angular.ts:143]: Yes it is part of the yarn 3 documentation and this is only used in tests

**Pull request closed without merging**